### DISCUSSION TITLE
iOS 17.x:  Released App crashes during startup on iPhone with slightes changes

### DISCUSSION THREAD

**Discussion by MAUIoxo** [dotnet/maui#24340] [Q&A]
Hi all,

just wanted to ask if you guys are also having serious problems deploying your **Release** built App on your iPhone with **iOS 17.6**?

It started with an App that that could be run on my iPhone back in May. When I recently tried again to deploy it on my iPhone, just crashed on startup and did not tell anything I could use. It runs successfully on iPhone Simulator but with older iOS 17.2 (also sometimes on 17.5).

As I did not find the error, I tried to recreate my App by copying single files and deploying it. By doing this, I found out that slightest changed in the App, like updating or adding a new NuGet-Package for instance or some other files did abosolutely easily crash the App on Startup when deployed to my iPhone.

Regarding the NuGet-Package-Updates, I found out that clearing my NuGet-Caches, deleting all bin/obj folders in my Solution and maybe also clearing everything on my Mac related to my compiled App fixed it then. I could then deploy and run the App without a code change

Currently, I think this issue started when I switched to iOS 17.5 or so, but is definitively present all the time on iOS 17.6. It really makes development a pain in the ass as you almost never know what caused the problem and if there even is one.

Is this a known issue with this iOS? What could cause this and is there anything I can do and make it more stable again allowing the deployes without crash? Or is there anything I can/should add to my .csproj when working with this iOS version that makes it more stable? Working with MAUI 8.0.80 right now, but would not blame MAUI

**Comment by jarlem** [comment]
Does it work in debug mode? Have you tried to add UseInterpreter true in the PropertyGroup for IOS in csproj?
My app need this in order to work in release mode for IOS due to some libs I have included. Just my two cents.

**Comment by MAUIoxo** [comment]
It is working in **Debug** on iOS Simulator 17.2/17.5 and also sometimes was working with added changes as **Release** on iOS Simulator with iOS 17.5, but then crashed on my iPhone with iOS 17.6 on startup

According to the DevExpress Build Settings page from the Link below, I've set the `<UseInterpreter>false</UseInterpreter>` for **Release**. Should it be changed to `true` in general?

[https://docs.devexpress.com/MAUI/404979/common-concepts/build-configuration?utm_source=SupportCenter&utm_medium=website&utm_campaign=docs-feedback&utm_content=T1245248#devexpress-build-settings](https://docs.devexpress.com/MAUI/404979/common-concepts/build-configuration?utm_source=SupportCenter&utm_medium=website&utm_campaign=docs-feedback&utm_content=T1245248#devexpress-build-settings)

```xml
<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|AnyCPU'">
	<MtouchUseLlvm>false</MtouchUseLlvm>
	<UseInterpreter>true</UseInterpreter>
	<MtouchInterpreter>all</MtouchInterpreter>
</PropertyGroup>

<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|AnyCPU'">
	<MtouchUseLlvm>true</MtouchUseLlvm>
	<UseInterpreter>false</UseInterpreter>
	<MtouchInterpreter>all</MtouchInterpreter>
</PropertyGroup>
```

**Comment by jarlem** [comment]
It should not be set to true in general, but my experience is (depending on what libraries are used in the app) that it will not work without it in release mode. See this discussion: https://github.com/dotnet/maui/issues/13019

**Reply by MAUIoxo** [reply]
The interesting thing is that it is not like it is not working in general. As soon as I delete all my NuGet-Packages, my bin/obj folders and additionally also my App related folder on my Mac, it works afterwards also on my iPhone. This is the strange thing as there was no code change in between this. And a simple added oder updated NuGet can/will make this necessary. This is what confuses me the most
